tingbot/graphics.py

```diff
@@ -217,6 +217,51 @@ def text(self, string, xy=None, color='grey', align='center', font=None, font_si
 
         self.image(text_image, xy=xy, align=align, scale=1)
 
+    def oval(self, xy=None, size=(100,100), color='grey', align='center'):
+        """
+        Draws an oval.
+        
+        Args:
+            xy (tuple): The position (x, y) to draw the oval, as measured from the top-left.
+            size (tuple): The size (width, height) of the oval.
+            color (tuple or str): The color (r, g, b) or color name.
+            align (str): How to align the outside box of the oval relative to `xy`, or relative to the drawing surface
+                if `xy` is None. Defaults to 'center'.
+        """
+        if len(size) != 2:
+            raise ValueError('size should be a 2-tuple')
+
```

Comment:
To give nice error messages, we should check that the elements of `size` are numbers. But don't check using `type(el) is not int` - that would stop floating point numbers from getting through. Instead use `isinstance(el, numbers.Real)`.